starport/pkg/spn/spn.go

```diff
@@ -296,341 +164,72 @@ func (c *Client) broadcast(ctx context.Context, clientCtx client.Context, msgs .
-		ChainID:    chainID,
-		ProposalID: int32(id),
-	}
-	res, err := queryClient.ShowProposal(ctx, param)
-	if err != nil {
-		return Proposal{}, err
+	// make sure that account has enough balances before broadcasting.
+	if err := c.makeSureAccountHasTokens(ctx, clientCtx.GetFromAddress().String()); err != nil {
+		return err
 	}
 
-	return c.toProposal(*res.Proposal)
-}
-
-// ProposalOption configures Proposal to set a spesific type of proposal.
-type ProposalOption func(*Proposal)
-
-// AddAccountProposal creates an add account proposal option.
-func AddAccountProposal(address string, coins types.Coins) ProposalOption {
-	return func(p *Proposal) {
-		p.Account = &ProposalAddAccount{address, coins}
-	}
-}
+	c.out.Reset()
 
-// AddValidatorProposal creates an add validator proposal option.
-func AddValidatorProposal(gentx jsondoc.Doc, validatorAddress string, selfDelegation types.Coin, p2pAddress string) ProposalOption {
-	return func(p *Proposal) {
-		p.Validator = &ProposalAddValidator{gentx, validatorAddress, selfDelegation, p2pAddress}
-	}
+	return nil
 }
 
-// Propose proposes given proposals in batch for chainID by using SPN accountName.
-func (c *Client) Propose(ctx context.Context, accountName, chainID string, proposals ...ProposalOption) error {
-	if len(proposals) == 0 {
-		return errors.New("at least one proposal required")
-	}
-
-	clientCtx, err := c.buildClientCtx(accountName)
-	if err != nil {
+// broadcast directly broadcasts the messages into spn handlers
+func (c *Client) broadcast(ctx context.Context, clientCtx client.Context, msgs ...types.Msg) error {
+	if err := c.prepareBroadcast(ctx, clientCtx, msgs...); err != nil {
 		return err
 	}
 
-	var msgs []types.Msg
-
-	for _, p := range proposals {
-		var proposal Proposal
-		p(&proposal)
-
-		switch {
-		case proposal.Account != nil:
-			addr, err := types.AccAddressFromBech32(proposal.Account.Address)
-			if err != nil {
-				return err
-			}
-
-			// Create the proposal payload
-			payload := genesistypes.NewProposalAddAccountPayload(
-				addr,
-				proposal.Account.Coins,
-			)
-
-			msgs = append(msgs, genesistypes.NewMsgProposalAddAccount(
-				chainID,
-				clientCtx.GetFromAddress(),
-				payload,
-			))
-
-		case proposal.Validator != nil:
-			// Get the validator address
-			addr, err := types.AccAddressFromBech32(proposal.Validator.ValidatorAddress)
-			if err != nil {
-				return err
-			}
-			validatorAddress := types.ValAddress(addr)
-
-			// Create the proposal payload
-			payload := genesistypes.NewProposalAddValidatorPayload(
-				proposal.Validator.Gentx,
-				validatorAddress,
-				proposal.Validator.SelfDelegation,
-				proposal.Validator.P2PAddress,
-			)
-
-			msgs = append(msgs, genesistypes.NewMsgProposalAddValidator(
-				chainID,
-				clientCtx.GetFromAddress(),
-				payload,
-			))
+	// broadcast tx.
+	if err := tx.BroadcastTx(clientCtx, c.factory, msgs...); err != nil {
+		if strings.Contains(err.Error(), "not found") {
```

Comment:
Maybe we can put this inside a `handleBroadcastError()` to reuse in the other place as well.

So it can be used like this:
```go
if err := tx.BroadcastTx(clientCtx, c.factory, msgs...); handleBroadcastError(err) != nil {}
```

```go
func handleBroadcastError(err error) error {
    if err != nil {
        return nil
    }
    if strings.Contains...
}
```